Documentation/teaching/so2/assign1-kprobe-based-tracer.rst

```diff
@@ -0,0 +1,173 @@
+==================================
+Assignment 1 - Kprobe based tracer
+==================================
+
+-  Deadline: :command:`Wednesday, April 7th 2021, 23:00`
+
+Assignment's Objectives
+=======================
+
+*  acquiring knowledge related to the instrumentation of functions in the Linux kernel (``kretprobes`` mechanism)
+*  gaining knowledge regarding the ``/proc`` file system from the Linux kernel
+*  getting more familiar with the data structures specific to the Linux kernel (``hash table``, ``list``)
+
```

Comment:
I would replace `acuiqiring` with `aquire`, `gaining` with `gain`, etc. 

Also, last bullet I would rephrase it like this:
* get familiar with data structure specific to Linux kernel (``hash_table`` and ``list``).

Reply:
Done.